### DISCUSSION TITLE
WiFi problems with MCUfriend shield & UNO style Wemos R32?

### DISCUSSION THREAD

**Discussion by mprowe** [Bodmer/TFT_eSPI#1775] [Q&A - General] [answered]
I have the MCUfriend shield stacked on top of a Wemos R32 (just like the comments on Intro section for this project).
Without the shield in place, the WiFi modes (WIFI_STA, WIFI_AP & WIFI_STA_AP)  work just fine.
However, with the display shield in place, Serial.print() diagnostics suggest that the WiFi is working just fine.
But there is no WiFi connection (in or out) possible from an external device?

Can anyone confirm or reject the possibility that the display is generating RF noise that is jamming the ESP32's WiFi signals?

Regards, M.

**Comment by Bodmer** [comment, marked as answer]
The TFT typically has a metal backplate. The proximity probably detunes the WiFi antenna and causes an antiphase cancellation reflection. This reduces range.
Options are to move the screen further away from antenna with extra headers or use an external leaded antenna.

Another possibility, I have 2 ESP32 boards that seem to have a silicon level manufacture bug/defect and refuse to connect to the router sometimes, the odd thing is that just adding debug print statements will seem to cure the problem sometimes! Another seemingly identical board is totally reliable!

**Comment by mprowe** [comment]
Thank you for your comments Bodmar,
I think you are right about the metal backplate. I have now removed the ESP32 module with the onboard antenna and fitted a 32U.
So the project is now back on course.
Regards, M.
